<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RandomStringClass</class>
 <widget class="QWidget" name="RandomStringClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RandomString</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="MainLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Character set:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="CharacterSetField">
         <property name="text">
          <string>abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890</string>
         </property>
         <property name="placeholderText">
          <string>Character set</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>String length:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="LengthBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000000</number>
           </property>
           <property name="value">
            <number>16</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Strings:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="StringNumberBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="GenerateButton">
         <property name="text">
          <string>Generate</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="OutputField">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>Output</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>GenerateButton</sender>
   <signal>clicked()</signal>
   <receiver>RandomStringClass</receiver>
   <slot>OnGenerateButtonPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CharacterSetField</sender>
   <signal>textChanged(QString)</signal>
   <receiver>RandomStringClass</receiver>
   <slot>OnCharacterSetChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LengthBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>RandomStringClass</receiver>
   <slot>OnStringLengthChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnGenerateButtonPress()</slot>
  <slot>OnCharacterSetChange()</slot>
  <slot>OnStringLengthChange()</slot>
 </slots>
</ui>
